backend/apps/webui/main.py

```diff
@@ -243,33 +246,23 @@ def get_pipe_id(form_data: dict) -> str:
     return pipe_id
 
 
-def get_function_params(function_module, form_data, user, extra_params={}):
+def get_function_params(function_module, form_data, user, extra_params=None):
+    if extra_params is None:
+        extra_params = {}
+
     pipe_id = get_pipe_id(form_data)
     # Get the signature of the function
     sig = inspect.signature(function_module.pipe)
-    params = {"body": form_data}
-
-    for key, value in extra_params.items():
-        if key in sig.parameters:
-            params[key] = value
-
-    if "__user__" in sig.parameters:
-        __user__ = {
-            "id": user.id,
-            "email": user.email,
-            "name": user.name,
-            "role": user.role,
-        }
-
-        try:
-            if hasattr(function_module, "UserValves"):
-                __user__["valves"] = function_module.UserValves(
-                    **Functions.get_user_valves_by_id_and_user_id(pipe_id, user.id)
-                )
-        except Exception as e:
-            print(e)
+    addition_params = {k: v for k, v in extra_params.items() if k in sig.parameters}
+    params = {"body": form_data} | addition_params
 
-        params["__user__"] = __user__
+    if "__user__" in params and hasattr(function_module, "UserValves"):
+        user_valves = Functions.get_user_valves_by_id_and_user_id(pipe_id, user.id)
+        if user_valves:
```

Comment:
This should be reverted, default user valves should be provided if user_valves aren't set.

Reply:
```py
if hasattr(function_module, "UserValves"):
        __user__["valves"] = function_module.UserValves(
            **Functions.get_user_valves_by_id_and_user_id(pipe_id, user.id)
        )
```

Reply:
My bad: I meant `is not None` since `get_user_valves_by_id_and_user_id` returns None instead of throwing an exception.